ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java

```diff
@@ -133,15 +138,30 @@ public QueryPublisher createQueryPublisher(
     if (statement.getStatement().isPullQuery()) {
       final ImmutableAnalysis analysis = ksqlEngine
           .analyzeQueryWithNoOutputTopic(statement.getStatement(), statement.getStatementText());
-      return createTablePullQueryPublisher(
-          analysis,
-          context,
-          serviceContext,
-          statement,
-          pullQueryMetrics,
-          workerExecutor,
-          metricsCallbackHolder
-      );
+      final DataSource dataSource = analysis.getFrom().getDataSource();
+      final DataSource.DataSourceType dataSourceType = dataSource.getDataSourceType();
+      switch (dataSourceType) {
+        case KTABLE:
+          return createTablePullQueryPublisher(
+              analysis,
+              context,
+              serviceContext,
+              statement,
+              pullQueryMetrics,
+              workerExecutor,
+              metricsCallbackHolder
+          );
+        case KSTREAM:
+          throw new KsqlStatementException(
+              "Pull queries are not supported on streams.",
+              statement.getStatementText()
+          );
+        default:
+          throw new KsqlStatementException(
+              "Unexpected data source type for pull query: " + dataSourceType,
+              statement.getStatementText()
+          );
+      }
```

Comment:
This is the main contribution of this PR: Detect the source type up front and add a switch statement so that we'll have a place to add the stream pull query in later.